ui/components/app/snaps/snap-ui-renderer/components/asset-selector.test.ts

```diff
@@ -0,0 +1,222 @@
+            keyring: {
+              type: 'Snap Keyring',
+            },
+          },
+        },
+      },
+      balances: {
+        '8c33fc18-6c52-44b1-b8fa-550b934a05ef': {
+          'solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp/slip44:105': {
+            amount: '1',
+            unit: 'SOL',
+          },
+          'solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp/token:EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v':
+            {
+              amount: '2',
+              unit: 'USDC',
+            },
+        },
+      },
+      accountsAssets: {
+        '8c33fc18-6c52-44b1-b8fa-550b934a05ef': [
+          'solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp/slip44:105',
+          'solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp/token:EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v',
+        ],
+      },
+      assetsMetadata: {
+        'solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp/slip44:105': {
+          fungible: true,
+          iconUrl:
+            'https://static.cx.metamask.io/api/v2/tokenIcons/assets/solana/5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp/slip44/501.png',
+          name: 'Solana',
+          symbol: 'SOL',
+          units: [
+            {
+              decimals: 9,
+              name: 'Solana',
+              symbol: 'SOL',
+            },
+          ],
+        },
+        'solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp/token:EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v':
+          {
+            fungible: true,
+            iconUrl:
+              'https://static.cx.metamask.io/api/v2/tokenIcons/assets/solana/5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp/token/EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v.png',
+            name: 'USDC',
+            symbol: 'USDC',
+            units: [
+              {
+                decimals: 9,
+                name: 'USDC',
+                symbol: 'USDC',
+              },
+            ],
+          },
+      },
+      conversionRates: {
+        'solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp/slip44:105': {
+          conversionTime: 1745405595549,
+          currency: 'swift:0/iso4217:USD',
+          expirationTime: 1745409195549,
+          rate: '151.36',
+        },
+        'solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp/token:EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v':
+          {
+            conversionTime: 1745405595549,
+            currency: 'swift:0/iso4217:USD',
+            expirationTime: 1745409195549,
+            rate: '1.00',
+          },
+      },
+    },
+  };
+  it('renders an asset selector', () => {
+    const { container } = renderInterface(
+      Box({
+        children: AssetSelector({
+          name: 'asset-selector',
+          addresses: [
+            'solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp:7S3P4HxJpyyigGzodYwHtCxZyUQe9JiBMHyRWXArAaKv',
+          ],
+        }),
+      }),
+      {
+        state: {
+          'asset-selector': {
+            asset: 'solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp/slip44:105',
+            name: 'Solana',
+            symbol: 'SOL',
+          },
+        },
```

Comment:
Do you mind pulling these into their own variables and re-using them across the tests?